src/store/modules/conversations.js

```diff
@@ -97,6 +105,26 @@ export default {
         await conversationsAPI.toggleEmailNotifications(conversationId, value)
         commit('updateEmailNotifications', { conversationId, value })
       },
+
+      /**
+       * Add reaction to a message.
+       */
+      async editReaction ({ state, commit }, { action, name, messageId, userId }) {
```

Comment:
Maybe you don't need to pass the user id? Rather always assume that the current user did it  -> rootGetters['auth/userId']

Also, I wouldn't pass the action, rather make it a `toggleReaction` - the vuex module has all information to find out if the reaction already exists and which endpoint to call.

Generally, I'd keep the interface between vuex and components as small as possible and strip logic from components. Vuex modules are easier to test and easier to reuse.